src/utils/parser/utils.js

```diff
@@ -228,24 +255,60 @@ function getExpression(
 
   traverse(ast, {
     enter(path) {
-      const node = path.node;
-      if (
-        t.isMemberExpression(node) &&
-        node.property.name === token &&
-        nodeContainsLocation({ node, location })
-      ) {
-        const expr = getMemberExpression(node);
-        expression = {
-          value: expr.join("."),
-          location: node.loc,
-        };
+      if (!expression) {
+        expression = resolveExpression(path, token, location);
       }
     },
   });
 
   return expression;
 }
 
+// Resolves a token (at location) in the source to determine if it is in scope
+// of the given frame and the expression (if any) to which it belongs
+function resolveToken(
+  source: SourceText,
+  token: string,
+  location: Location,
+  frame: Frame
+): ?TokenResolution {
+  const ast = getAst(source);
+  const scopes = [];
+  let expression = null;
+
+  if (isEmpty(ast)) {
+    return;
+  }
+
+  traverse(ast, {
+    enter(path) {
+      // if we haven't found an expression yet, determine if the token is part
+      // of one
+      if (!expression) {
```

Comment:
There's no need to continue looking for an `expression` if we've found one, right? Maybe I'm misunderstanding how the AST traversal happens, though.